core-bundle/assets/controllers/scroll-controller.js

```diff
@@ -0,0 +1,51 @@
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+    static classes = ['up', 'down']
+
+    initialize () {
+        this.lastScroll = 0;
+        this.onScroll = this.onScroll.bind(this);
+    }
+
+    connect () {
+        window.addEventListener('scroll', this.onScroll, { passive: true });
+    }
+
+    disconnect () {
+        window.removeEventListener('scroll', this.onScroll);
+    }
+
+    up () {
+        if (this.hasUpClass) {
+            document.body.classList.add(this.upClass);
+        }
+
+        if (this.hasDownClass) {
+            document.body.classList.remove(this.downClass);
+        }
+    }
+
+    down () {
+        if (this.hasUpClass) {
+            document.body.classList.remove(this.upClass);
+        }
+
+        if (this.hasDownClass) {
+            document.body.classList.add(this.downClass);
+        }
+    }
+
+    onScroll () {
+        // Make sure the scroll value is between 0 and maxScroll
+        const currentScroll = Math.max(0, Math.min(document.documentElement.scrollHeight - document.documentElement.clientHeight, window.scrollY));
+
+        if (this.lastScroll < currentScroll) {
+            this.down();
+        } else if (this.lastScroll > currentScroll) {
+            this.up();
+        }
+
+        this.lastScroll = currentScroll;
```

Comment:
```suggestion
        if (this.lastScroll < currentScroll - 20) {
            this.down();
            this.lastScroll = currentScroll;
        } else if (this.lastScroll > currentScroll + 20 || currentScroll < 20) {
            this.up();
            this.lastScroll = currentScroll;
        }
```

Should we add a little threshold so that scrolling by very small amounts does not trigger this?

Reply:
That doesn‘t work for me. The ` || currentScroll < 20` part adds the `up` class even though I scroll down.

Reply:
And the `up` class does not get removed when scrolling down less than 20 pixels.

Reply:
Then we should use `|| currentScroll < 1` (to make sure scrolling completely to the top shows the header).

Reply:
Added in 713fc202cc6423cff2529e5d441ffab304368659.